services/users/src/routes/profile.ts

```diff
@@ -7,15 +8,48 @@ export const profileRoutes = express.Router({ mergeParams: true });
 
 profileRoutes.route("/").get(
   asyncHandler(async (req, res) => {
-    const profile = await ProfileModel.findOne({
-      user: req.params.userId,
-    });
+    const limit = parseInt(req.query.limit as string);
+    const offset = parseInt(req.query.offset as string);
+    const regex = (req.query.regex as string) === "true";
+
+    let re;
+    if (regex) {
+      const search = (req.query.search as string).split(/\s+/).join("");
+      re = new RegExp(search, "i");
+    } else {
+      re = new RegExp(req.query.search as string);
+    }
 
-    res.send(profile || {});
+    const matchCount = await ProfileModel.find({
+      $or: [
+        { "name.first": { $regex: re } },
+        { "name.middle": { $regex: re } },
+        { "name.last": { $regex: re } },
+        { phoneNumber: { $regex: re } },
+      ],
+    }).count();
+    console.log(matchCount);
+    const profiles = await ProfileModel.find({
+      $or: [
+        { "name.first": { $regex: re } },
+        { "name.middle": { $regex: re } },
+        { "name.last": { $regex: re } },
+        { phoneNumber: { $regex: re } },
+      ],
+    })
+      .skip(offset)
+      .limit(limit);
+
+    return res.status(200).json({
+      offset: offset + profiles.length,
+      total: matchCount,
+      count: profiles.length,
+      profiles: profiles,
+    });
   })
 );
 
-profileRoutes.route("/").post(
+profileRoutes.route("/:userId").post(
```

Comment:
I don't think we need this change here? We should already be getting their userId